extensions/amp-render/1.0/test/test-amp-render.js

```diff
@@ -305,5 +315,48 @@ describes.realWin(
       expect(ampScript.getImpl).to.have.been.calledOnce;
       expect(fetchJsonStub).not.to.have.been.called;
     });
+
+    it('should use the specified xssi-prefix and key attributes', async () => {
+      const json = {
+        fullName: {
+          firstName: 'Joe',
+          lastName: 'Biden',
+        },
+      };
+
+      const fetchJsonStub = env.sandbox
+        .stub(BatchedJsonModule, 'batchFetchJsonFor')
+        .callThrough();
+
+      env.sandbox.stub(Services, 'batchedXhrFor').returns({
+        fetchJson: () => Promise.resolve(json),
+      });
+
+      env.sandbox.stub(Services, 'xhrFor').returns({
+        fetch: () => Promise.resolve(json),
+        xssiJson: () => Promise.resolve(json),
+      });
```

Comment:
Are all 3 of these stubs necessary?

Reply:
Yes. `fetchJsonStub` is used to verify the params to the function (hence the `callThrough`). The other 2 are to mimic the respective methods so we can verify that the component renders some content.